File: CodeGenDotNet 159059-1372 (a)/VBCode/Tools/MetadataExtraction/IExtractMetaData.vb
' ====================================================================
'  Copyright ©2004, Kathleen Dollard, All Rights Reserved.
' ====================================================================
'   I'm distributing this code so you'll be able to use it to see code
'   generation in action and I hope it will be useful and you'll enjoy 
'   using it. This code is provided "AS IS" without warranty, either 
'   expressed or implied, including implied warranties of merchantability 
'   and/or fitness for a particular purpose. 
' ====================================================================
'  Summary: Interface for all metadata extraction tools

Option Strict On
Option Explicit On 

Imports System
Imports System.Data
Imports System.Diagnostics

'! Class Summary: Interface for metadata extracton 

Public Interface IExtractMetaData
   Property UseVerboseNames() As Boolean

   Property UseProcContents() As Boolean

   Property ServerName() As String

   Function CreateMetaData( _
               ByVal skipStoredProcs As Boolean, _
               ByVal setSelectPatterns As String, _
               ByVal selectPatterns As String, _
               ByVal removePrefix As String, _
               ByVal lookupPrefix As String, _
               ByVal ParamArray databaseNames() As String) _
               As Xml.XmlDocument


End Interface

File: CodeGenDotNet 159059-1372 (a)/CSharp/Tools/MetadataExtraction/IExtractMetaData.cs
// ====================================================================
//  Copyright ©2004, Kathleen Dollard, All Rights Reserved.
// ====================================================================
//   I'm distributing this code so you'll be able to use it to see code
//   generation in action and I hope it will be useful and you'll enjoy 
//   using it. This code is provided "AS IS" without warranty, either 
//   expressed or implied, including implied warranties of merchantability 
//   and/or fitness for a particular purpose. 
// ====================================================================
//  Summary: Interface for all metadata extraction tools

using System;
using System.Data;
using System.Diagnostics;

namespace KADGen.Metadata
{
	/// <summary>
	/// Interface for metadata extracton
	/// </summary>
	public interface IExtractMetaData
	{
		bool UseVerboseNames
		{
			get;
			set;
		}

		bool UseProcContents
		{
			get;
			set;
		}

		string ServerName
		{
			get;
			set;
		}

		System.Xml.XmlDocument CreateMetaData(	bool skipStoredProcs, 
			string setSelectPatterns,
			string selectPatterns,
			string removePrefix,
			string lookupPrefix,
			params string[] databaseNames );
	}
}
